### ISSUE TITLE
[RLlib] MultivariateNormal distributions in bandits do not actually update sample variance

### ISSUE THREAD

**Issue by ajaffe-lyft** [ray-project/ray#29543]
### What happened + What you expected to happen

For the conclusion first, it looks like BanditLinUCB and BanditLinTS are both broken because the variance of the parameter estimates appear to be fixed to 1 at initialization. 

Investigation ahead!

A teammate and I were experimenting with using BanditLinTS against a toy data example we created with arms with fixed reward ratios.

As we ran the experiment, we found that the internal estimates of the arms looked correct. That is to say, as time went on, the covariance matrix (just a single value) shrank, and the value of theta was approaching the correct value.

Still, we were super surprised to find that for some reason, the model wasn't really reliably picking one arm over another.

With that in mind, I tried hitting up the internal distributions of the arms directly to see what the sample function returns. Turns out the means are correct, but empirically the variances for both arms are 1.

<img width="1093" alt="Screen Shot 2022-10-21 at 1 27 18 AM" src="https://user-images.githubusercontent.com/58490020/197158899-947f4cf2-83ba-4392-b7b6-b14966eb8706.png">

I figured that was really strange, so I messed around with initializing the MultivariateNormal distributions on my own but couldn't reproduce the bug.

I then went and checked the `.__dict__` attribute of the arms returned and noticed something weird. There was a hidden `_unbroadcasted_scale_tril` variable that's 1!

```
{'covariance_matrix': tensor([[0.0002]]),
 'loc': tensor([0.3006]),
 '_batch_shape': torch.Size([]),
 '_event_shape': torch.Size([1]),
 '_unbroadcasted_scale_tril': tensor([[1.]])}
```
That matched my observation.

Looking into torch's source code, it looks like they actually use the `_unbroadcasted_scale` parameter for their sampling: https://github.com/pytorch/pytorch/blob/master/torch/distributions/multivariate_normal.py#L210

So, my guess was that covariance does update, but it doesn't change the internal private variable. And that was correct!

<img width="1104" alt="Screen Shot 2022-10-21 at 2 19 50 AM" src="https://user-images.githubusercontent.com/58490020/197161486-9e2b0e1d-f254-4794-a1c9-d18a2844bb08.png">

Lastly, I noticed that the covariance is actually [passed through a cholesky decomp ](https://github.com/pytorch/pytorch/blob/master/torch/distributions/multivariate_normal.py#L155) when setting that internal variable.

When I repeated the experiment by setting the scale_tril variable instead, the update was indeed reflected in the internal state! Notice how the `_unbroadcasted_scale_tril` variable goes from tiny to huge.

<img width="1035" alt="Screen Shot 2022-10-21 at 2 27 48 AM" src="https://user-images.githubusercontent.com/58490020/197163119-71ff4e39-bbb5-43f7-b422-8874b01187aa.png">

So, it looks like `torch.linalg.cholesky` screws up the whole update propagation thing and you're stuck with the variance you initialized at, which is 1

If you've got two arms whose true values are within 1 of each other (i.e. tons of bernoulli variables) you aren't going to pick a good arm much more than a bad one.

Perhaps even worse, for an experiment where you've got large expected values for the arms that also have variances large relative to 1 (e.g, two arms with true means of 1000, and 1050, true variances of 100 each) You could end up in a situation where you get an slightly unlucky draw on that 1050 arm which plops your mean estimate for it at 950, and then you could just keep drawing from the 1000-valued arm and never get a sample mean near enough to 950 for your fixed 1-variance uncertainties to overlap.

### Versions / Dependencies

Python3.8,
 ray==2.0.0,
torch==1.12.1
Ubuntu 18.04.6 LTS

### Reproduction script
```python
import gym
import numpy as np
from gym.spaces import Discrete, Box
from matplotlib import pyplot as plt
from ray.rllib.algorithms.bandit.bandit import BanditLinTS
from ray.rllib.env import EnvContext
TRAINING_ITERATIONS = 100
ARM_COUNT = 2
BETTER_ARM_PROBABILITY = 0.3
ACTION_TO_REWARD_PROBABILITY = {
    0: 0.1,
    1: BETTER_ARM_PROBABILITY
}
class NonContextualBanditEnv(gym.Env):
    def __init__(self, config: EnvContext):
        self.action_space = Discrete(ARM_COUNT)
        # actually no features - can remove?
        self.observation_space = Box(0.0, 1.0, shape=(1,), dtype=np.float32)
    def reset(self):
        return [1.0]
    def step(self, action):
        reward = np.random.binomial(1.0, ACTION_TO_REWARD_PROBABILITY[action])
        return [1.0], reward, True, {"regret": BETTER_ARM_PROBABILITY - ACTION_TO_REWARD_PROBABILITY[action]}
    
if __name__ == "__main__":
    """
    Test how well the model converges to the better arm that has a higher reward probability.
    """
    algo = BanditLinTS(env=NonContextualBanditEnv, config={'framework': 'torch'})
    policy = algo.get_policy()
    model = policy.model
    arm_means = []
    training_results = []
    reward_per_round = []
    # Log progress for each training iteration
    for i in range(TRAINING_ITERATIONS):
        training_results.append(algo.train())
        arm_means.append([float(model.arms[i].theta) for i in range(ARM_COUNT)])
        reward_per_round.append(sum([algo.compute_single_action(observation=[1.0]) for _ in range(100)]))

print(model.arms[1].dist)
n = 100000
samples = [float(model.arms[1].dist.sample()) for _ in range(n)]
print(np.mean(samples), np.var(samples))
print(model.arms[1].dist.__dict__)
```

### Issue Severity

High: It blocks me from completing my task.

**Comment by kouroshHakha** [comment]
Hello, @ajaffe-lyft Thanks for filing this issue. The issue has been addressed in [this](https://github.com/ray-project/ray/pull/29666) PR. Here is the updated mean and standard deviation as the training progresses. 
<img width="1273" alt="image" src="https://user-images.githubusercontent.com/31483498/197901247-c8b239c0-43da-4b66-abc0-dace8d3868e7.png">
